tempesta_fw/http_sess.c

```diff
@@ -593,6 +628,76 @@ tfw_http_sess_obtain(TfwHttpReq *req)
 	return 0;
 }
 
+/**
+ * Get last used connection to reach server group @sg. Even if no messsages in
+ * this http session were explicitly forwarded to @sg, there still can be
+ * known connections to @sg if it appear to be a back-up connection for some
+ * other server group.
+ */
+TfwConnection *
+tfw_http_sess_get_conn(TfwHttpReq *req, TfwSrvGroup *sg)
+{
+	TfwHttpSess *sess = req->sess;
+	TfwHttpSessConns *s_conn;
+	TfwConnection *conn = NULL, *backup_conn = NULL;
+
+	if (!sess)
+		return NULL;
+
+	read_lock(&sess->conns_lock);
+
+	list_for_each_entry(s_conn, &sess->conns, list) {
+		if (s_conn->sg == sg) {
+			conn = s_conn->conn;
+			break;
+		}
+		else if (s_conn->conn &&
+			 (((TfwServer *)s_conn->conn->peer)->sg == sg))
+		{
+			backup_conn = s_conn->conn;
```

Comment:
But the search still continues? I am not sure I understand the logic. I would appreciate an explanation.